src/viewProviders/OnlineDevicesViewProvider.ts

```diff
@@ -49,7 +51,80 @@ export class OnlineDevicesViewProvider implements vscode.TreeDataProvider<vscode
 
     private devices: Array<RokuDeviceDetails>;
 
-    getChildren(element?: DeviceTreeItem | DeviceInfoTreeItem): vscode.ProviderResult<DeviceTreeItem[] | DeviceInfoTreeItem[]> {
+    private async getAppData(deviceIp: string): Promise<vscode.QuickPickItem[]> {
+        return new Promise<vscode.QuickPickItem[]>((resolve, reject) => {
+            const appNames: vscode.QuickPickItem[] = [];
+            const client: Socket = new Socket();
+            const port = 8060;
+
+            const httpRequest = `GET /query/apps HTTP/1.1\r\n` +
+                                `Host: ${deviceIp}:${port}\r\n` +
+                                `Connection: close\r\n\r\n`;
+            client.connect({ host: deviceIp, port: port }, () => {
+                client.write(httpRequest);
+            });
+
+            let responseData = '';
+            client.on('data', (data) => {
+                responseData += data.toString();
+            });
+
+            client.on('end', () => {
+                try {
+                    const parsedData = this.parseXmlResponse(responseData);
+                    resolve(parsedData);
+                } catch (error) {
+                    console.error('Error:', error);
+                    reject(error);
+                } finally {
+                    client.end();
+                }
+            });
+
+            client.on('error', (error) => {
+                console.error(`Error: ${error.message}`);
+                client.destroy();
+                reject(error);
+            });
+        });
+    }
```

Comment:
This seems fairly verbose. We already imported a request library in this project. You could do something like this instead:

in util.ts, add this code:
```typescript
import * as request from 'postman-request';

export class Util {
   // ...
    public httpGet(url: string) {
        return new Promise<Response>((resolve, reject) => {
            request.get(url, (err, response) => {
                return err ? reject(err) : resolve(response);
            });
        });
    }
   ///...
}
```

Then you can just call:

```typescript
const data = await util.httpGet(`${deviceIp}/query/apps`);
```

Reply:
ok, changes made